test/powershell/Language/Scripting/ActionPreference.Tests.ps1

```diff
@@ -0,0 +1,124 @@
+            }
+            catch {}
+             
+            It '$err.Count' { $err.Count | Should Be 1 }
+            It '$err[0] should not be $null' { $err[0] | Should Not Be $null }
+            It '$err[0].GetType().Name' { $err[0].GetType().Name | Should Be ActionPreferenceStopException }
+            It '$err[0].ErrorRecord' { $err[0].ErrorRecord | Should not BeNullOrEmpty }
+            It '$err[0].ErrorRecord.Exception.GetType().Name' { $err[0].ErrorRecord.Exception.GetType().Name | Should Be ItemNotFoundException }
+        }
+        
+        It 'ActionPreference Ignore Works' {            
+            $errorCount = $error.Count
+            Get-Process -Name asdfasdfsadfsadf -ErrorAction Ignore            
+
+            $error.Count | Should Be $errorCount
+        }
+        
+        Context 'action preference of Ignore cannot be set as a preference variable' {           
+            try {
+                $GLOBAL:errorActionPreference = "Ignore"
+                Get-Process -Name asdfasdfasdf
+                Throw "Execution OK"
+             } catch {
+                     $_.CategoryInfo.Reason | Should Be NotSupportedException
+             }
+        }
+
+        It 'action preference of Suspend cannot be set as a preference variable' {            
+            try { 
+                    $GLOBAL:errorActionPreference = "Suspend"
+                    Get-Process -Name asdfasdfasdf
+                    Throw "Execution OK"
+                } catch { 
+                    $_.CategoryInfo.Reason | Should Be ArgumentTransformationMetadataException
+                }
+        }
+
+        It 'enum disambiguation works' {
+            $errorCount = $error.Count
+            Get-Process -Name asdfasdfsadfsadf -ErrorAction Ig
+
+            $error.Count | Should Be $errorCount
+        }
+
+        It 'ErrorAction = Suspend works on Workflow' -Skip:$IsCoreCLR {
+           . .\TestsOnWinFullOnly.ps1
+            Run-TestOnWinFull "ActionPreference:ErrorAction=SuspendOnWorkflow"
+        }
+
+        It 'ErrorAction = Suspend does not work on functions' {
+            function MyHelperFunction {
+                [CmdletBinding()]
+                param()        
+                "Hello"
+            }
+    
+            try
+            {
+                MyHelperFunction -ErrorAction Suspend
+                Throw "Execution OK"
+            } catch {
+                $_.FullyQualifiedErrorId | Should Be "ParameterBindingFailed,MyHelperFunction"
+            }
+        }
+
+        It 'ErrorAction = Suspend does not work on cmdlets' {
+            try
+            {
+                Get-Process -ErrorAction Suspend
+                Throw "Execution OK"
+            }
+            catch {
+                $_.FullyQualifiedErrorId | Should Be "ParameterBindingFailed,Microsoft.PowerShell.Commands.GetProcessCommand"
+            }
+        }
+
+        Context 'WarningAction = Suspend does not work' {
```

Comment:
fixed

---
In reply to: [76494914](https://github.com/powershell/powershell/pull/2086#discussion_r76494914) [](ancestors = 76494914)